packages/cdk8s-cli/src/import/codegen.ts

```diff
@@ -87,24 +91,33 @@ export function generateConstruct(typegen: TypeGenerator, def: ApiObjectDefiniti
       code.line(` * Defines a "${def.fqn}" API object`);
       code.line(' * @param scope the scope in which to define this object');
       code.line(' * @param id a scope-local name for the object');
-      code.line(' * @param props initialiation props');
+      code.line(' * @param props initialization props');
       code.line(' */');
 
-      const hasRequired = schema?.required && Array.isArray(schema.required) && schema.required.length > 0;
-      const defaultProps = hasRequired ? '' : ' = {}';
       code.openBlock(`public constructor(scope: Construct, id: string, props: ${propsTypeName}${defaultProps})`);
-      emitInitializerSuper();
+
+      code.line(`super(scope, id, ${constructName}.propsWithGVK(props));`);
 
       code.closeBlock();
     }
 
-    function emitInitializerSuper() {
+    function emitPropsWithGVK() {
+
+      code.line('/**');
+      code.line(` * Adds "${def.fqn}" kind and apiVersion to props`);
+      code.line(' * @param props initialization props');
+      code.line(' */');
+
+      code.openBlock(`public static propsWithGVK(props: ${propsTypeName}${defaultProps}): any`);
```

Comment:
Instead of a method which merges GVK into an arbitrary props objects, why not just emit something like this:

```ts
export class KubePod extends ... {
  public static readonly GVK: GroupVersionKind = {
    apiVersion: 'v1',
    kind: 'Pod'
  };
}
```

Then, the caller can merge that into anything they want.

Reply:
I added both here so we can get cleaner usage with propsWithGVK()